nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -210,8 +218,22 @@ static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
     static_cast<NNTrainerInference *>(*private_data);
   g_return_val_if_fail(prop && nntrainer && in_info && out_info, -EINVAL);
 
-  auto num_input = in_info->num_tensors;
-  g_return_val_if_fail(num_input != 0, -EINVAL);
+  auto model_inputs = nntrainer->getInputDimension();
+
+  /// if num_input is zero, we set a default input shape with batch of 1.
+  if (in_info->num_tensors == 0) {
+    ml_logi(
+      "num input is zero, regard it as a default dimension with batch of 1");
+    auto mutable_in_info = const_cast<GstTensorsInfo *>(in_info);
+    mutable_in_info->num_tensors = model_inputs.size();
+    for (unsigned int i = 0u; i < model_inputs.size(); ++i) {
+      auto default_dim = model_inputs[i];
+      default_dim.batch(1);
```

Comment:
this will override the batchsize set in ini? I wonder by batchsize has to be reset and not directly take what nntrainer provides. Note that nntrainer defaults batchsize to 1 if not specified.

Reply:
consider the scenario where model is trained with batchsize of n and saved for inference.

It will be sensible that for inference default batchsize is 1, even though batchsize is specified.

Reply:
Then let's add in ini documentation that the batch size in ini will be ignored in inference.

Reply:
agreed :)